.github/workflows/test-and-publish-nightly.yml

```diff
@@ -65,65 +136,58 @@ jobs:
           architecture: ${{ matrix.architecture }}
       - name: Checkout sources
         uses: actions/checkout@v4
-      - name: Build and test
-        id: build
+      - name: Set artifact name
+        id: set_artifact_name
+        run: echo "artifact_name=async-profiler-${GITHUB_SHA:0:7}-${{ matrix.asprof-binaries-job || matrix.runson.display }}" >> $GITHUB_OUTPUT
+        shell: bash
+        env:
+          GITHUB_SHA: ${{ github.sha }}
+      - name: Download async-profiler release artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ steps.set_artifact_name.outputs.artifact_name }}
+          path: async_profiler_release
+      - name: Download async-profiler JAR artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: async-profiler-jars
+          path: jar_artifacts
+      - name: Extract async-profiler artifact
+        id: extract_artifact
         run: |
-          HASH=$(echo ${{ github.sha }} | cut -c-7)
+          release_archive=$(basename $(find async_profiler_release -type f -iname "async-profiler-*" ))
```

Comment:
Can we build the name instead of using `find`, to ensure that we are getting the intended file?

Reply:
The reasoning here is that the name of the `.tar.gz` is decided by `Makefile`, which we do not fully control in GitHub Actions. So, we would need one of the following:
- impose the full name via some variable set in GHA
- figure what the name is going to look like, and make sure changes in `Makefile` are reflected in GHA when the logic to create the name changes in the former

Both are less clean than using `find` in my opinion, especially because the directory `async_profiler_release` should only contain one file (the `.tar.gz`)